### PR TITLE
colormap mpl examples

### PR DESCRIPTION
**Pull request by 0ctagon**

### CODE REVIEW

**Review by cyrraz — COMMENTED** [review]
For 3 days and 3 nights, I have been thinking to this PR. I found that it does not only go against the first principles of `plothist`, but it is also a threat to the students we promised to protect and an obstacle to the progress of science. We shall resist.

**Comment by 0ctagon** [comment]
How can we add new colors then?

File: plothist/plothist_style.py [hunk 1]
```diff
@@ -37,6 +39,116 @@ def set_style(style="default"):
...
+
+    References
+    ----------
+    Matplotlib colormaps: https://matplotlib.org/stable/gallery/color/colormap_reference.html
+    """
+    plt_cmap = plt.get_cmap(cmap)
+
+    values = np.linspace(0, 1, N)
+
+    colors = [plt_cmap(value) for value in values]
```
Comment (cyrraz) [inline plothist/plothist_style.py:147]: Can't you do `plt_cmap(values)`?

File: docs/usage/style.rst [hunk 2]
```diff
@@ -12,12 +12,12 @@ Default style
...
 
-Color palette
-=============
+Color palettes
+==============
 
 When displaying quantities such as yields or intensities, it is recommended to utilize Perceptually Uniform Sequential Colormaps like the cubehelix colormap for accurate representation.
 
-We provide the ``cubehelix_palette()`` (see `here <https://plothist.readthedocs.io/en/latest/documentation/documentation.html#plothist.plotters.cubehelix_palette>`_) function to create such colormap.
+We provide the ``cubehelix_palette()`` (see `here <https://plothist.readthedocs.io/en/latest/documentation/documentation.html#plothist.plotters.cubehelix_palette>`_) function to create such colormap. All the examples in the documentation are using this colorpalette.
```
Comment (cyrraz) [inline docs/usage/style.rst:20]: color palette

File: docs/usage/style.rst [hunk 3]
```diff
@@ -26,7 +26,25 @@ We provide the ``cubehelix_palette()`` (see `here <https://plothist.readthedocs.
...
 
 .. image:: ../img/adv_cubehelix.svg
-   :alt: Presentation style
+   :alt: Cubehelix example
+..    :width: 500
+
+
+We also provide the ``get_cmap_palette(cmap, N)``, which can take any ``cmap`` matplotlib colorpalette and sequence it in ``N`` different colors (see `here <https://matplotlib.org/stable/gallery/color/colormap_reference.html>`_ for the different ``cmap`` names).
+
+Here are a few palette that we recommand using:
```
Comment (cyrraz) [inline docs/usage/style.rst:35]: palettes

File: docs/usage/style.rst [hunk 4]
```diff
@@ -26,7 +26,25 @@ We provide the ``cubehelix_palette()`` (see `here <https://plothist.readthedocs.
     palette = cubehelix_palette(11)
 
 .. image:: ../img/adv_cubehelix.svg
-   :alt: Presentation style
+   :alt: Cubehelix example
+..    :width: 500
+
+
+We also provide the ``get_cmap_palette(cmap, N)``, which can take any ``cmap`` matplotlib colorpalette and sequence it in ``N`` different colors (see `here <https://matplotlib.org/stable/gallery/color/colormap_reference.html>`_ for the different ``cmap`` names).
```
Comment (cyrraz) [inline docs/usage/style.rst:33]: color palette

File: docs/img/usage_colorpalette_examples.svg
Comment (cyrraz) [inline docs/img/usage_colorpalette_examples.svg:1]: Lets recommend less palettes.
My proposal:
* viridis_r
* coolwarm_r
* YlGnBu
* Accent

File: docs/usage/style.rst [hunk 5]
```diff
@@ -9,26 +9,117 @@ Default style
...
+You can also use the ``cubehelix_palette()`` function to tweak the colormap parameters.
+
+
+Matplotlib palettes
+-------------------
+
+
+The function ``get_color_palette(cmap, N)`` can also take any ``cmap`` matplotlib color palette and sequence it in ``N`` different colors (see `here <https://matplotlib.org/stable/gallery/color/colormap_reference.html>`_ for the different ``cmap`` names).
+
+We strongly recommand using ``viridis``, ``coolwarm`` or ``YlGnBu_r``:
```
Comment (cyrraz) [inline docs/usage/style.rst:93]: remove strongly. Less is more.

File: plothist/plothist_style.py [hunk 6]
```diff
@@ -37,6 +41,134 @@ def set_style(style="default"):
...
+    """
+    if N < 2:
+        raise ValueError("The number of colors asked should be >1.")
+
+    if cmap == "ggplot":
+        if N > 7:
+            warnings.warn(
+                f"Only 7 colors are available in the default style cycle ({N} asked). Will return 7 colors.",
+            )
+            N = 7
```
Comment (cyrraz) [inline plothist/plothist_style.py:158]: No, if the user wants more than 7 colors, we should raise a value error, not a warning.

**Review by cyrraz — APPROVED** [review]

**Pull request merged**